compilation error: sample_hevc_fei

The following compilation error occurs (I'm using GCC 7.2.1):
```
[ 94%] Building CXX object samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/src/pipeline_hevc_fei.cpp.o
cd /home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release/samples/sample_hevc_fei && /usr/bin/c++  -DLINUX -DLINUX32 -DLINUX64 -DLINUX_TARGET_PLATFORM -DLINUX_TARGET_PLATFORM_BDW -DMFX_FILE_VERSION=\"0.18.1.26\" -DMFX_PRODUCT_VERSION=\"0.0.000.0000\" -DMSDK_BUILD=\"\" -DUNIX -D_FILE_OFFSET_BITS=64 -D__USE_LARGEFILE64 -I/home/uaeoff/Work/workspace/media/build/msdk/api/mediasdk_structures -I/home/uaeoff/Work/workspace/media/build/msdk/api/include -I/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/../sample_common/include -I/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/include  -msse4.2 -std=c++11 -m64 -g -pipe -fPIC -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wall -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-unused  -Wformat -Wformat-security -DNDEBUG   -I/usr/include/libdrm -I/home/uaeoff/Work/workspace/media/install/include -I/home/uaeoff/Work/workspace/media/install/include -I/home/uaeoff/Work/workspace/media/install/include  -DMFX_VA -DLIBVA_SUPPORT -DLIBVA_DRM_SUPPORT -DLIBVA_X11_SUPPORT   -I/home/uaeoff/Work/workspace/media/install/include -I/usr/include/libdrm   -DLIBVA_SUPPORT -DLIBVA_DRM_SUPPORT -DLIBVA_X11_SUPPORT   -Werror  -o CMakeFiles/sample_hevc_fei.dir/src/pipeline_hevc_fei.cpp.o -c /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/pipeline_hevc_fei.cpp
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/pipeline_hevc_fei.cpp: In member function ‘void CEncodingPipeline::PrintInfo()’:
/home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/pipeline_hevc_fei.cpp:122:6: error: ‘__builtin___snprintf_chk’ output may be truncated before the last format character [-Werror=format-truncation=]
 void CEncodingPipeline::PrintInfo()
      ^~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:862:0,
                 from /usr/include/c++/7/cstdio:42,
                 from /usr/include/c++/7/ext/string_conversions.h:43,
                 from /usr/include/c++/7/bits/basic_string.h:6347,
                 from /usr/include/c++/7/string:52,
                 from /usr/include/c++/7/bits/locale_classes.h:40,
                 from /usr/include/c++/7/bits/ios_base.h:41,
                 from /usr/include/c++/7/ios:42,
                 from /usr/include/c++/7/ostream:38,
                 from /usr/include/c++/7/iostream:39,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/../sample_common/include/sample_defs.h:24,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/include/sample_hevc_fei_defs.h:24,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/include/pipeline_hevc_fei.h:22,
                 from /home/uaeoff/Work/workspace/media/build/msdk/samples/sample_hevc_fei/src/pipeline_hevc_fei.cpp:20:
/usr/include/bits/stdio2.h:65:44: note: ‘__builtin___snprintf_chk’ output between 2 and 6 bytes into a destination of size 5
        __bos (__s), __fmt, __va_arg_pack ());
                                            ^
cc1plus: all warnings being treated as errors
make[2]: *** [samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/build.make:186: samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/src/pipeline_hevc_fei.cpp.o] Error 1
make[2]: Leaving directory '/home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release'
make[1]: *** [CMakeFiles/Makefile2:1959: samples/sample_hevc_fei/CMakeFiles/sample_hevc_fei.dir/all] Error 2
make[1]: Leaving directory '/home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release'
make: *** [Makefile:133: all] Error 2
make: Leaving directory '/home/uaeoff/Work/workspace/media/build/msdk/__cmake/intel64.make.release'
```

The `samples/sample_hevc_fei/CMakeLists.txt` file hard-codes `-Werror` in compiler flags.  Instead of hard-coding, we should let `WARNING_FLAGS` determine whether to use `-Werror` or not.

--- a/samples/sample_hevc_fei/CMakeLists.txt
+++ b/samples/sample_hevc_fei/CMakeLists.txt
@@ -17,7 +17,7 @@ include_directories (
 list( APPEND LIBS_VARIANT sample_common )
 
 set(DEPENDENCIES ${skipping} itt libmfx dl pthread)
-set( defs "-Werror" )
+set( defs "${WARNING_FLAGS}" )
 make_executable( shortname universal "nosafestring" )
 
 install( TARGETS ${target} RUNTIME DESTINATION ${MFX_SAMPLES_INSTALL_BIN_DIR} )

